File: scala/classic_puzzles_easy/TheDescent.scala
// https://www.codingame.com/training/easy/the-descent/solution

import math._
import scala.util._

object Player extends App {

    // game loop
    while(true) {
        val mountain = (0 to 7).map(_ => readInt)
                               .toList
                               .zipWithIndex
                               .maxBy(_._1)._2
                
        println(mountain) // The index of the mountain to fire on.
    }
}

File: go/classic_puzzles_easy/TheDescent.go
// https://www.codingame.com/training/easy/the-descent

package main

import "fmt"

func main() {
	mountains := make([]int, 8)
	for {
		for i := 0; i < 8; i++ {
			fmt.Scan(&mountains[i])
		}

		max := mountains[0]
		index := 0
		for i, h := range mountains {
			if h > max {
				max = h
				index = i
			}
		}

		fmt.Println(index)
	}
}
